private/bufpkg/bufwkt/cmd/wkt-go-data/main.go

```diff
@@ -0,0 +1,324 @@
+	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmodulebuild"
+	"github.com/bufbuild/buf/private/pkg/app"
+	"github.com/bufbuild/buf/private/pkg/app/appcmd"
+	"github.com/bufbuild/buf/private/pkg/app/appflag"
+	"github.com/bufbuild/buf/private/pkg/protosource"
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"github.com/bufbuild/buf/private/pkg/storage/storageos"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	programName = "wkt-go-data"
+	pkgFlagName = "package"
+	sliceLength = math.MaxInt64
+)
+
+func main() {
+	appcmd.Main(context.Background(), newCommand())
+}
+
+func newCommand() *appcmd.Command {
+	flags := newFlags()
+	builder := appflag.NewBuilder(programName)
+	return &appcmd.Command{
+		Use:  fmt.Sprintf("%s path/to/google/protobuf/include", programName),
+		Args: cobra.ExactArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appflag.Container) error {
+				return run(ctx, container, flags)
+			},
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	Pkg string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(
+		&f.Pkg,
+		pkgFlagName,
+		"",
+		"The name of the generated package.",
+	)
+}
+
+func run(ctx context.Context, container appflag.Container, flags *flags) error {
+	dirPath := container.Arg(0)
+	packageName := flags.Pkg
+	if packageName == "" {
+		packageName = filepath.Base(dirPath)
+	}
+	readWriteBucket, err := storageos.NewProvider(storageos.ProviderWithSymlinks()).NewReadWriteBucket(dirPath)
+	if err != nil {
+		return err
+	}
+	pathToData, err := getPathToData(ctx, readWriteBucket)
+	if err != nil {
+		return err
+	}
+	protosourceFiles, err := getProtosourceFiles(ctx, container, readWriteBucket)
+	if err != nil {
+		return err
+	}
+	fullNameToMessage, err := protosource.FullNameToMessage(protosourceFiles...)
+	if err != nil {
+		return err
+	}
+	fullNameToEnum, err := protosource.FullNameToEnum(protosourceFiles...)
+	if err != nil {
+		return err
+	}
+	golangFileData, err := getGolangFileData(
+		pathToData,
+		fullNameToMessage,
+		fullNameToEnum,
+		packageName,
+	)
+	if err != nil {
+		return err
+	}
+	_, err = container.Stdout().Write(golangFileData)
+	return err
+}
+
+func getPathToData(ctx context.Context, bucket storage.ReadBucket) (map[string][]byte, error) {
+	pathToData := make(map[string][]byte)
+	if err := storage.WalkReadObjects(
+		ctx,
+		bucket,
+		"",
+		func(readObject storage.ReadObject) error {
+			data, err := io.ReadAll(readObject)
+			if err != nil {
+				return err
+			}
+			pathToData[readObject.Path()] = data
+			return nil
+		},
+	); err != nil {
+		return nil, err
+	}
+	return pathToData, nil
+}
+
+func getProtosourceFiles(
+	ctx context.Context,
+	container appflag.Container,
+	bucket storage.ReadBucket,
+) ([]protosource.File, error) {
+	// TODO: why is this not working? this is the path that should be used, not NewModuleForBucket
```

Comment:
I am aware that this todo is here but I do not know what the difference between `bufmodule.NewModuleForBucket` and `bufmodulebuild.NewModuleBucketBuilder` is